geomesa-kafka/geomesa-kafka-utils/src/main/scala/org/locationtech/geomesa/kafka/RecordVersions.scala

```diff
@@ -54,6 +56,15 @@ object RecordVersions extends LazyLogging {
     }
   }
 
+  private val _getTimestamp: ConsumerRecord[_, _] => Option[Long] = {
+    consumerMethods.find(m => m.getName == "timestamp" && m.getParameterCount == 0).map{ method =>
+      record: ConsumerRecord[_,_] => Option(method.invoke(record).asInstanceOf[java.lang.Long]).map(Long2long)
```

Comment:
I think the timestamp returns a primitive Long (at least in 2.0), so you don't need to cast to java.lang.Long and unbox

Reply:
Actually, could we make this just return `Long` and not `Option[Long]`, and return `System.currentTimeMillis` for unsupported versions? That would match the previous behavior, and we don't *really* care about old kafka versions enough to handle missing timestamps I'd say